synapse/storage/databases/main/relations.py

```diff
@@ -765,6 +765,41 @@ def _get_if_user_has_annotated_event(txn: LoggingTransaction) -> bool:
             "get_if_user_has_annotated_event", _get_if_user_has_annotated_event
         )
 
+    async def get_mutual_event_relations(
+        self, event: EventBase
+    ) -> Set[Tuple[str, str, str]]:
+        """
+        Fetch event meta data for events which related to the same event as the given event.
+
+        If the given event has no relation information, returns an empty set.
+
+        Args:
+            event: The event to fetch relations for.
+
+        Returns:
+            A set of tuples of:
+                The relation type
+                The sender
+                The event type
+        """
+        relation = relation_from_event(event)
+        if not relation:
+            return set()
```

Comment:
Maybe it would make sense to push this onto the caller so we can cache the result?

Reply:
I did this and added a cache. I think this is clearer. 👍